[Automation] native-image run fails for org.junit.platform:junit-platform-commons:1.8.2

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.8.2" ./gradlew clean test -Pcoordinates="org.junit.platform:junit-platform-commons:1.8.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581280
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
data/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:+AllowJRTFileSystem@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=3cc2eb51-f650-d257-9087-d1df208206c8 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'org.junit.platform.junit-platform-commons_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.1s @ 0.15GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:+AllowJRTFileSystem' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (35.8s @ 0.85GB)
    8,216 types,   9,313 fields, and  36,201 methods found reachable
    2,676 types,     213 fields, and   1,618 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (4.4s @ 0.96GB)
[4/8] Parsing methods...      [**]                                                                       (4.4s @ 0.69GB)
[5/8] Inlining methods...     [***]                                                                      (2.3s @ 0.77GB)
[6/8] Compiling methods...    [****]                                                                    (18.4s @ 0.90GB)
[7/8] Laying out methods...   [**]                                                                       (3.4s @ 0.86GB)
[8/8] Creating image...       [**]                                                                       (3.4s @ 0.99GB)
   9.45MB (35.60%) for code area:    22,462 compilation units
  13.43MB (50.62%) for image heap:  212,151 objects and 155 resources
   3.66MB (13.78%) for other data
  26.54MB in total image size, 24.51MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.64MB java.base                                            2.76MB byte[] for code metadata
   2.14MB svm.jar (Native Image)                               2.60MB byte[] for java.lang.String
 574.12kB assertj-core-3.22.0.jar                              1.35MB java.lang.String
 481.21kB java.xml                                             1.02MB java.lang.Class
 199.89kB junit-jupiter-engine-5.9.2.jar                     854.46kB com.oracle.svm.core.hub.DynamicHubCompanion
 139.34kB jdk.zipfs                                          529.68kB heap alignment
 121.94kB junit-platform-launcher-1.9.2.jar                  450.82kB java.util.concurrent.ConcurrentHashMap$Node
 107.69kB junit-platform-commons-1.9.2.jar                   430.19kB byte[] for general heap data
 103.19kB java.logging                                       336.42kB java.lang.Object[]
  99.83kB junit-platform-engine-1.9.2.jar                    300.86kB java.util.HashMap$Node
 491.68kB for 23 more packages                                 2.79MB for 2053 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 18 component(s) is embedded in binary (934B). 19 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        5.1s (6.2% of total time) in 720 GCs | Peak RSS: 1.77GB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile/org.junit.platform.junit-platform-commons_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.junit.platform.junit-platform-commons_tests-tests' in 1m 20s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_junit_platform.junit_platform_commons.AnnotationUtilsTest > findsPublicFieldsAnnotatedWithMetaAnnotation() SUCCESSFUL

org_junit_platform.junit_platform_commons.ClassLoaderUtilsTest > getsLocationForClassLoadedByApplicationClassLoader() SUCCESSFUL

org_junit_platform.junit_platform_commons.ClasspathScannerTest > scansPackageResourcesUsingDefaultClassLoader() SUCCESSFUL

org_junit_platform.junit_platform_commons.ModuleUtilsInnerModuleReferenceScannerTest > scansNamedModuleAndLoadsMatchingClass() FAILED

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > createsInstanceFromMatchingDeclaredConstructor() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > resolvesOutermostInstanceFromInnerClass() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > findsDeclaredConstructorsMatchingPredicate() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > getsPublicMethodByNameAndParameterTypes() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > findsMethodsDeclaredByInterfaces() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > loadsObjectArrayTypesBySourceName() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > findsFieldsDeclaredByImplementedInterfaces() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > readsPrivateFieldValueByName() SUCCESSFUL

org_junit_platform.junit_platform_commons.RuntimeUtilsTest > determinesDebugModeWithoutThrowing() SUCCESSFUL


Failures (1):
  JUnit Jupiter:ModuleUtilsInnerModuleReferenceScannerTest:scansNamedModuleAndLoadsMatchingClass()
    MethodSource [className = 'org_junit_platform.junit_platform_commons.ModuleUtilsInnerModuleReferenceScannerTest', methodName = 'scansNamedModuleAndLoadsMatchingClass', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       java.base@25.0.3/jdk.internal.module.SystemModuleFinders$SystemImage.reader(SystemModuleFinders.java:125)
       java.base@25.0.3/jdk.internal.module.SystemModuleFinders$ModuleContentSpliterator.<init>(SystemModuleFinders.java:491)
       java.base@25.0.3/jdk.internal.module.SystemModuleFinders$SystemModuleReader.list(SystemModuleFinders.java:466)
       org.junit.platform.commons.util.ModuleUtils$ModuleReferenceScanner.scan(ModuleUtils.java:176)
       org.junit.platform.commons.util.ModuleUtils.scan(ModuleUtils.java:152)
       [...]
     Caused by: java.lang.NullPointerException
       java.base@25.0.3/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.3/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       java.base@25.0.3/jdk.internal.jimage.ImageReaderFactory.<clinit>(ImageReaderFactory.java:66)
       [...]

Test run finished after 6 ms
[         8 containers found      ]
[         0 containers skipped    ]
[         8 containers started    ]
[         0 containers aborted    ]
[         8 containers successful ]
[         0 containers failed     ]
[        13 tests found           ]
[         0 tests skipped         ]
[        13 tests started         ]
[         0 tests aborted         ]
[        12 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date
May 12, 2026 11:01:43 AM org.junit.platform.commons.util.ClasspathScanner findClassesForUri
WARNING: Error scanning files for URI resource:/org_junit_platform/junit_platform_commons
java.nio.file.FileSystemNotFoundException: The Native Image Resource File System is not present. Please create a new file system using the `newFileSystem` operation before attempting any file system operations on resource URIs.
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.resources.NativeImageResourceFileSystemProvider.getFileSystem(NativeImageResourceFileSystemProvider.java:144)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.resources.NativeImageResourceFileSystemProvider.getPath(NativeImageResourceFileSystemProvider.java:155)
	at java.base@25.0.3/java.nio.file.Path.of(Path.java:206)
	at java.base@25.0.3/java.nio.file.Paths.get(Paths.java:95)
	at org.junit.platform.commons.util.CloseablePath.create(CloseablePath.java:58)
	at org.junit.platform.commons.util.ClasspathScanner.findClassesForUri(ClasspathScanner.java:108)
	at org.junit.platform.commons.util.ClasspathScanner.lambda$findClassesForUris$0(ClasspathScanner.java:100)
	at java.base@25.0.3/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:214)
	at java.base@25.0.3/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1716)
	at java.base@25.0.3/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
	at java.base@25.0.3/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
	at java.base@25.0.3/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base@25.0.3/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base@25.0.3/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:723)
	at org.junit.platform.commons.util.ClasspathScanner.findClassesForUris(ClasspathScanner.java:103)
	at org.junit.platform.commons.util.ClasspathScanner.scanForClassesInPackage(ClasspathScanner.java:83)
	at org.junit.platform.commons.util.ReflectionUtils.findAllClassesInPackage(ReflectionUtils.java:1043)
	at org.junit.platform.commons.util.ReflectionUtils.findAllClassesInPackage(ReflectionUtils.java:1036)
	at org.junit.platform.commons.support.ReflectionSupport.findAllClassesInPackage(ReflectionSupport.java:138)
	at org_junit_platform.junit_platform_commons.ClasspathScannerTest.scansPackageResourcesUsingDefaultClassLoader(ClasspathScannerTest.java:23)
	at java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.3/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.3/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile/org.junit.platform.junit-platform-commons_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 23s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.junit.platform:junit-platform-commons:1.8.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 25s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.8.2 at 2026-05-12T11:01:44Z with exit code 1 after 86s
❌ ERROR: 'native-image run' for 1.8.2 failed with exit code 1.
FAILED[native-image run][1.8.2][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.8.2" ./gradlew clean test -Pcoordinates="org.junit.platform:junit-platform-commons:1.8.2"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.8.2" ./gradlew clean test -Pcoordinates="org.junit.platform:junit-platform-commons:1.8.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581291
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
x-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.junit.platform.junit-platform-commons_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.7s @ 0.17GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:+AllowJRTFileSystem' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (35.9s @ 0.84GiB)
    8,358 types,   9,406 fields, and  36,783 methods found reachable
      305 types,      94 fields, and     752 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      155 resource accesses registered with 263.73KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.3s @ 0.96GiB)
[4/8] Parsing methods...      [**]                                                                      (4.2s @ 0.66GiB)
[5/8] Inlining methods...     [***]                                                                     (2.0s @ 0.73GiB)
[6/8] Compiling methods...    [****]                                                                   (19.9s @ 0.74GiB)
[7/8] Laying out methods...   [**]                                                                      (3.0s @ 0.91GiB)
[8/8] Creating image...       [**]                                                                      (3.4s @ 1.04GiB)
   8.41MiB (33.18%) for code area:    22,834 compilation units
  13.25MiB (52.28%) for image heap:  217,088 objects and 24 resources
   3.68MiB (14.54%) for other data
  25.34MiB in total image size, 23.38MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
 955.47KiB o.g.n.runtime/com.oracle.svm.core.code              2.61MiB byte[] for code metadata
 708.33KiB java.base/java.util                                 2.58MiB byte[] for string data
 408.29KiB java.base/jdk.internal.classfile.impl               1.37MiB java.base/java.lang.String
 297.42KiB java.base/java.lang                                 0.99MiB java.base/java.lang.Class
 228.80KiB java.base/java.text                               914.16KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 224.56KiB java.base/com.sun.crypto.provider                 556.41KiB byte[] for general heap data
 224.48KiB java.base/java.util.concurrent                    556.30KiB heap alignment
 223.43KiB org.assertj.core.internal.bytebuddy.jar.asm       327.63KiB java.base/java.lang.Object[]
 149.71KiB java.base/java.io                                 324.06KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 149.52KiB java.base/java.lang.invoke                        305.27KiB byte[] for reflection metadata
   4.92MiB for 313 more packages                               2.79MiB for 2063 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (962B). 19 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.4s (6.5% of total time) in 709 GCs | Peak RSS: 1.64GiB | CPU load: 3.59
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile/org.junit.platform.junit-platform-commons_tests-tests (executable, 23.38MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'org.junit.platform.junit-platform-commons_tests-tests' in 1m 21s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_junit_platform.junit_platform_commons.AnnotationUtilsTest > findsPublicFieldsAnnotatedWithMetaAnnotation() SUCCESSFUL

org_junit_platform.junit_platform_commons.ClassLoaderUtilsTest > getsLocationForClassLoadedByApplicationClassLoader() SUCCESSFUL

org_junit_platform.junit_platform_commons.ClasspathScannerTest > scansPackageResourcesUsingDefaultClassLoader() SUCCESSFUL
May 12, 2026 11:00:30 AM org.junit.platform.commons.util.ClasspathScanner findClassesForUri
WARNING: Error scanning files for URI resource:/org_junit_platform/junit_platform_commons
java.nio.file.FileSystemNotFoundException: The Native Image Resource File System is not present. Please create a new file system using the `newFileSystem` operation before attempting any file system operations on resource URIs.
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.resources.NativeImageResourceFileSystemProvider.getFileSystem(NativeImageResourceFileSystemProvider.java:146)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.resources.NativeImageResourceFileSystemProvider.getPath(NativeImageResourceFileSystemProvider.java:157)
	at java.base@25.0.2/java.nio.file.Path.of(Path.java:206)
	at java.base@25.0.2/java.nio.file.Paths.get(Paths.java:95)
	at org.junit.platform.commons.util.CloseablePath.create(CloseablePath.java:58)
	at org.junit.platform.commons.util.ClasspathScanner.findClassesForUri(ClasspathScanner.java:108)
	at org.junit.platform.commons.util.ClasspathScanner.lambda$findClassesForUris$0(ClasspathScanner.java:100)
	at java.base@25.0.2/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:214)
	at java.base@25.0.2/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1716)
	at java.base@25.0.2/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)

	at java.base@25.0.2/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
	at java.base@25.0.2/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base@25.0.2/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base@25.0.2/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:723)
	at org.junit.platform.commons.util.ClasspathScanner.findClassesForUris(ClasspathScanner.java:103)
	at org.junit.platform.commons.util.ClasspathScanner.scanForClassesInPackage(ClasspathScanner.java:83)
	at org.junit.platform.commons.util.ReflectionUtils.findAllClassesInPackage(ReflectionUtils.java:1043)
	at org.junit.platform.commons.util.ReflectionUtils.findAllClassesInPackage(ReflectionUtils.java:1036)
	at org.junit.platform.commons.support.ReflectionSupport.findAllClassesInPackage(ReflectionSupport.java:138)
	at org_junit_platform.junit_platform_commons.ClasspathScannerTest.scansPackageResourcesUsingDefaultClassLoader(ClasspathScannerTest.java:23)
	at java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.2/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile/org.junit.platform.junit-platform-commons_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 25s
org_junit_platform.junit_platform_commons.ModuleUtilsInnerModuleReferenceScannerTest > scansNamedModuleAndLoadsMatchingClass() FAILED

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > createsInstanceFromMatchingDeclaredConstructor() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > resolvesOutermostInstanceFromInnerClass() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > findsDeclaredConstructorsMatchingPredicate() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > getsPublicMethodByNameAndParameterTypes() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > findsMethodsDeclaredByInterfaces() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > loadsObjectArrayTypesBySourceName() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > findsFieldsDeclaredByImplementedInterfaces() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > readsPrivateFieldValueByName() SUCCESSFUL

org_junit_platform.junit_platform_commons.RuntimeUtilsTest > determinesDebugModeWithoutThrowing() SUCCESSFUL


Failures (1):
  JUnit Jupiter:ModuleUtilsInnerModuleReferenceScannerTest:scansNamedModuleAndLoadsMatchingClass()
    MethodSource [className = 'org_junit_platform.junit_platform_commons.ModuleUtilsInnerModuleReferenceScannerTest', methodName = 'scansNamedModuleAndLoadsMatchingClass', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       java.base@25.0.2/jdk.internal.module.SystemModuleFinders$SystemImage.reader(SystemModuleFinders.java:129)
       java.base@25.0.2/jdk.internal.module.SystemModuleFinders$ModuleContentSpliterator.<init>(SystemModuleFinders.java:491)
       java.base@25.0.2/jdk.internal.module.SystemModuleFinders$SystemModuleReader.list(SystemModuleFinders.java:466)
       org.junit.platform.commons.util.ModuleUtils$ModuleReferenceScanner.scan(ModuleUtils.java:176)
       org.junit.platform.commons.util.ModuleUtils.scan(ModuleUtils.java:152)
       [...]
     Caused by: java.lang.NullPointerException
       java.base@25.0.2/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.2/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       java.base@25.0.2/jdk.internal.jimage.ImageReaderFactory.<clinit>(ImageReaderFactory.java:66)
       [...]

Test run finished after 5 ms
[         8 containers found      ]
[         0 containers skipped    ]
[         8 containers started    ]
[         0 containers aborted    ]
[         8 containers successful ]
[         0 containers failed     ]
[        13 tests found           ]
[         0 tests skipped         ]
[        13 tests started         ]
[         0 tests aborted         ]
[        12 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.junit.platform:junit-platform-commons:1.8.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 27s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.8.2 at 2026-05-12T11:00:30Z with exit code 1 after 87s
❌ ERROR: 'native-image run' for 1.8.2 failed with exit code 1.
FAILED[native-image run][1.8.2][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.8.2" ./gradlew clean test -Pcoordinates="org.junit.platform:junit-platform-commons:1.8.2"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.8.2" ./gradlew clean test -Pcoordinates="org.junit.platform:junit-platform-commons:1.8.2"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25727047580/job/75542581411
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
dtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.junit.platform.junit-platform-commons_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.8s @ 0.18GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:+AllowJRTFileSystem' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.28GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (38.8s @ 0.92GiB)
    8,867 types,  10,181 fields, and  39,771 methods found reachable
      433 types,      94 fields, and     951 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      157 resource accesses registered with 263.87KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.3s @ 1.01GiB)
[4/8] Parsing methods...      [**]                                                                      (4.6s @ 0.74GiB)
[5/8] Inlining methods...     [***]                                                                     (2.1s @ 0.77GiB)
[6/8] Compiling methods...    [*****]                                                                  (21.6s @ 0.81GiB)
[7/8] Laying out methods...   [**]                                                                      (3.1s @ 0.98GiB)
[8/8] Creating image...       [**]                                                                      (3.8s @ 1.10GiB)
   9.52MiB (33.39%) for code area:    25,146 compilation units
  15.06MiB (52.82%) for image heap:  251,894 objects and 26 resources
   3.93MiB (13.79%) for other data
  28.52MiB in total image size, 26.38MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.06MiB o.g.n.runtime/com.oracle.svm.core.code              2.96MiB byte[] for code metadata
 714.68KiB java.base/java.util                                 2.80MiB byte[] for string data
 408.29KiB java.base/jdk.internal.classfile.impl               1.48MiB java.base/java.lang.String
 304.97KiB java.base/java.lang                                 1.06MiB java.base/java.lang.Class
 267.01KiB java.base/com.sun.crypto.provider                 969.83KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 231.36KiB java.base/java.text                               912.48KiB byte[] for general heap data
 224.60KiB java.base/java.util.concurrent                    514.58KiB heap alignment
 223.43KiB org.assertj.core.internal.bytebuddy.jar.asm       372.94KiB byte[] for reflection metadata
 192.93KiB java.base/sun.security.provider                   361.97KiB java.base/java.lang.Object[]
 173.14KiB java.base/sun.security.x509                       343.95KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
   5.79MiB for 323 more packages                               3.37MiB for 2208 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (962B). 19 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       6.0s (6.8% of total time) in 823 GCs | Peak RSS: 1.71GiB | CPU load: 3.63
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile/org.junit.platform.junit-platform-commons_tests-tests (executable, 26.38MiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'org.junit.platform.junit-platform-commons_tests-tests' in 1m 27s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

May 12, 2026 11:04:52 AM org.junit.platform.commons.util.ClasspathScanner findClassesForUri
WARNING: Error scanning files for URI resource:/org_junit_platform/junit_platform_commons
java.nio.file.FileSystemNotFoundException: The Native Image Resource File System is not present. Please create a new file system using the `newFileSystem` operation before attempting any file system operations on resource URIs.
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.resources.NativeImageResourceFileSystemProvider.getFileSystem(NativeImageResourceFileSystemProvider.java:146)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.resources.NativeImageResourceFileSystemProvider.getPath(NativeImageResourceFileSystemProvider.java:157)
	at java.base@25.0.2/java.nio.file.Path.of(Path.java:206)
	at java.base@25.0.2/java.nio.file.Paths.get(Paths.java:95)
	at org.junit.platform.commons.util.CloseablePath.create(CloseablePath.java:58)
	at org.junit.platform.commons.util.ClasspathScanner.findClassesForUri(ClasspathScanner.java:108)
	at org.junit.platform.commons.util.ClasspathScanner.lambda$findClassesForUris$0(ClasspathScanner.java:100)
	at java.base@25.0.2/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:214)
	at java.base@25.0.2/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1716)
	at java.base@25.0.2/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
	at java.base@25.0.2/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
	at java.base@25.0.2/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base@25.0.2/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base@25.0.2/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:723)
	at org.junit.platform.commons.util.ClasspathScanner.findClassesForUris(ClasspathScanner.java:103)
	at org.junit.platform.commons.util.ClasspathScanner.scanForClassesInPackage(ClasspathScanner.java:83)
	at org.junit.platform.commons.util.ReflectionUtils.findAllClassesInPackage(ReflectionUtils.java:1043)
	at org.junit.platform.commons.util.ReflectionUtils.findAllClassesInPackage(ReflectionUtils.java:1036)
	at org.junit.platform.commons.support.ReflectionSupport.findAllClassesInPackage(ReflectionSupport.java:138)
	at org_junit_platform.junit_platform_commons.ClasspathScannerTest.scansPackageResourcesUsingDefaultClassLoader(ClasspathScannerTest.java:23)
	at java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
	at java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.2/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:820)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:796)

org_junit_platform.junit_platform_commons.AnnotationUtilsTest > findsPublicFieldsAnnotatedWithMetaAnnotation() SUCCESSFUL

org_junit_platform.junit_platform_commons.ClassLoaderUtilsTest > getsLocationForClassLoadedByApplicationClassLoader() SUCCESSFUL

org_junit_platform.junit_platform_commons.ClasspathScannerTest > scansPackageResourcesUsingDefaultClassLoader() SUCCESSFUL

org_junit_platform.junit_platform_commons.ModuleUtilsInnerModuleReferenceScannerTest > scansNamedModuleAndLoadsMatchingClass() FAILED

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > createsInstanceFromMatchingDeclaredConstructor() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > resolvesOutermostInstanceFromInnerClass() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > findsDeclaredConstructorsMatchingPredicate() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > getsPublicMethodByNameAndParameterTypes() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > findsMethodsDeclaredByInterfaces() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > loadsObjectArrayTypesBySourceName() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > findsFieldsDeclaredByImplementedInterfaces() SUCCESSFUL

org_junit_platform.junit_platform_commons.ReflectionUtilsTest > readsPrivateFieldValueByName() SUCCESSFUL

org_junit_platform.junit_platform_commons.RuntimeUtilsTest > determinesDebugModeWithoutThrowing() SUCCESSFUL


Failures (1):
  JUnit Jupiter:ModuleUtilsInnerModuleReferenceScannerTest:scansNamedModuleAndLoadsMatchingClass()
    MethodSource [className = 'org_junit_platform.junit_platform_commons.ModuleUtilsInnerModuleReferenceScannerTest', methodName = 'scansNamedModuleAndLoadsMatchingClass', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       java.base@25.0.2/jdk.internal.module.SystemModuleFinders$SystemImage.reader(SystemModuleFinders.java:129)
       java.base@25.0.2/jdk.internal.module.SystemModuleFinders$ModuleContentSpliterator.<init>(SystemModuleFinders.java:491)
       java.base@25.0.2/jdk.internal.module.SystemModuleFinders$SystemModuleReader.list(SystemModuleFinders.java:466)
       org.junit.platform.commons.util.ModuleUtils$ModuleReferenceScanner.scan(ModuleUtils.java:176)
       org.junit.platform.commons.util.ModuleUtils.scan(ModuleUtils.java:152)
       [...]
     Caused by: java.lang.NullPointerException
       java.base@25.0.2/java.util.Objects.requireNonNull(Objects.java:220)
       java.base@25.0.2/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
       java.base@25.0.2/jdk.internal.jimage.ImageReaderFactory.<clinit>(ImageReaderFactory.java:66)
       [...]

Test run finished after 5 ms
[         8 containers found      ]
[         0 containers skipped    ]
[         8 containers started    ]
[         0 containers aborted    ]
[         8 containers successful ]
[         0 containers failed     ]
[        13 tests found           ]
[         0 tests skipped         ]
[        13 tests started         ]
[         0 tests aborted         ]
[        12 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build/native/nativeTestCompile/org.junit.platform.junit-platform-commons_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 31s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.junit.platform:junit-platform-commons:1.8.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 32s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.8.2 at 2026-05-12T11:04:52Z with exit code 1 after 93s
❌ ERROR: 'native-image run' for 1.8.2 failed with exit code 1.
FAILED[native-image run][1.8.2][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.8.2" ./gradlew clean test -Pcoordinates="org.junit.platform:junit-platform-commons:1.8.2"]
```

--- a/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build.gradle
+++ b/tests/src/org.junit.platform/junit-platform-commons/1.8.2/build.gradle
@@ -15,6 +15,11 @@ dependencies {
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 
+tasks.named("nativeTest") {
+    runtimeArgs.add("-Djava.home=${System.getenv('JAVA_HOME')}")
+    runtimeArgs.add("-Djava.class.path=${sourceSets.test.runtimeClasspath.asPath}")
+}
+
 graalvmNative {
     binaries {
         test {

